agent/src/agentclient.rs

```diff
@@ -257,52 +258,137 @@ impl<T: APIServerClient> BrupopAgent<T> {
         Ok(())
     }
 
+    #[instrument(skip(self))]
+    async fn create_shadow_if_not_exist(&self) -> Result<()> {
```

Comment:
Nit:
```suggestion
    async fn create_shadow(&self) -> Result<()> {
```

I'm not sure if it necessary to include the condition in the function name. Unless there is a different `create_shadow` function that creates a shadow regardless of whether one already exists

Reply:
In the `run` loop, these two methods are called at the beginning of the loop. I feel with the condition in the name it would be more easy to understand.